index.js

```diff
@@ -51,6 +51,11 @@ export async function setupRuby(options = {}) {
   createGemRC(engine, version)
   envPreInstall()
 
+  // 1st conditional may need to change...
+  if (platform.startsWith('windows-202') && engine === 'jruby') {
```

Comment:
Could you do a better detection there, for instance one that checks if some path exists?

Reply:
How about
```js
  // JRuby can use compiled extension code, so make sure gcc exists.
  // As of Jan-2022, JRuby compiles against msvcrt.
  if (platform.startsWith('windows') && (engine === 'jruby') && 
    !fs.existsSync('C:\\msys64\\mingw64\\bin\\gcc.exe') {
    await require('./windows').installJRubyTools()
  }
```

Reply:
Yes that looks good, might just want to move that gcc path to `common` if it's used in multiple places

Reply:
I haven't added to common.  I think we should really try to minimize windows paths in index.js and common.js?  This being the one except, as I was unaware that people used extension code with FFI and JRuby.  BTW, does TruffleRuby build on Windows?

Reply:
OK.
>  BTW, does TruffleRuby build on Windows?

No it doesn't.